bind-operator/lib/charms/bind/v0/dns_record.py

```diff
@@ -244,19 +237,116 @@ class RequirerEntry(BaseModel):
         ttl: TTL.
         record_class: DNS record class.
         record_type: DNS record type.
-        record_data: the record value.
+        record_data: DNS record value (pydantic.IPvAnyAddress for A/AAAA, str otherwise).
         uuid: UUID for this entry.
     """
 
-    domain: str = Field(min_length=1)
-    host_label: str = Field(min_length=1)
-    ttl: Optional[int] = None
-    record_class: Optional[RecordClass] = None
-    record_type: Optional[RecordType] = None
-    record_data: IPvAnyAddress
+    domain: str = pydantic.Field(min_length=1)
+    host_label: str = pydantic.Field(min_length=1)
+    ttl: int
+    record_class: RecordClass = RecordClass.IN
+    record_type: RecordType
+    record_data: str | pydantic.IPvAnyAddress
     uuid: UUID
 
-    def validate_dns_entry(self, _: ValidationInfo) -> "RequirerEntry":
+    # Validator for record_data
+    @classmethod
+    @pydantic.field_validator("record_data")
+    def validate_record_data(
```

Comment:
In case of validation error, will the charm be in error status?

Reply:
The charm will have to decide what to do with the exception. I may have to revisit those parts in the charm (esp. bind)